### PR TITLE
SQL Backend Documentation

### PR DESCRIPTION
**Pull request by jimbishopp**
Documentation for SQL Backend #11048.

### CODE REVIEW

File: docs/pages/setup/reference/backends.mdx [hunk 1]
```diff
@@ -615,3 +615,65 @@ teleport:
...
+      client_cert_file: "/path/to/certs/client.dbuser.crt"
+      # Path to the database user's private key.
+      client_key_file: "/path/to/certs/client.dbuser.key"
+      # Path to the trusted certificate authority
+      # used to generate the client certificates.
+      ca_file: "/path/to/cockroachdb.cas"
+```
+
+Teleport will attempt to create a new database if the configured database does
+not exit. Creating a database requires `CREATEDB` privileges for the Teleport
```
Comment (r0mant) [inline docs/pages/setup/reference/backends.mdx:662]:
```suggestion
not exist. Creating a database requires `CREATEDB` privileges for the Teleport
```

**Review by r0mant — APPROVED** [review]

File: docs/pages/setup/reference/backends.mdx [hunk 2]
```diff
@@ -615,3 +615,65 @@ teleport:
...
+## PostgreSQL/CockroachDB (Preview)
+
+The PostgreSQL/CockroachDB backend supports storing Teleport cluster state
+in a self-hosted database instance.
+
+<Admonition
+  type="tip"
+  title="NOTE"
+>
+  The PostgreSQL/CockroachDB backend is currently in Preview mode.
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:628]:
```suggestion
  The PostgreSQL/CockroachDB backend is currently in Preview.
```

File: docs/pages/setup/reference/backends.mdx [hunk 3]
```diff
@@ -615,3 +615,65 @@ teleport:
...
+</Admonition>
+
+Connections require mutual TLS authentication. Be sure to configure your
+database instance to support certificate authentication before configuring
+the Teleport PostgreSQL/CockroachDB backend.
+See [PostgreSQL Certificate Authentication](https://www.postgresql.org/docs/current/auth-cert.html) to configure PostgreSQL
+or [Authenticating to CockroachDB Self-Hosted Clusters](https://www.cockroachlabs.com/docs/stable/authentication.html) to configure CockroachDB.
+
+Once your self-hosted database instance has been configured to support
+client authentication, configure all Teleport Auth servers to use the backend
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:638]:
```suggestion
certificate authentication, configure all Teleport Auth servers to use the backend
```
Is this what was meant here?

**Review by ptgott — APPROVED** [review]
Approved with two minor comments

**Pull request merged**